packages/react-events/src/Swipe.js

```diff
@@ -159,26 +240,21 @@ const SwipeResponder = {
           }
           const x = (obj: any).screenX;
           const y = (obj: any).screenY;
-          if (x < state.x && props.onSwipeLeft) {
-            state.direction = 3;
-          } else if (x > state.x && props.onSwipeRight) {
-            state.direction = 1;
+          if (x < state.x) {
```

Comment:
The logic here isn't quite right, as any vertical movement will cause the horizontal `direction` to be overwritten, even if the swipe is mainly along the horizontal axis. If a direction is to be reported, we need to determine whether it is "more" horizontal than vertical too.

Reply:
please check new implement, thank you.

Reply:
When a movement across the touch surface or screen should be considered a swipe. There are two variables at the distance traveled by the user's finger on the x or y-axis, and, the time it took. Based on these two factors, we can decide whether that action qualifies as a swipe and in what direction.

IMO we can have `distanceThreshold` props to weed out swipes and to be considered a swipe. 
For now, I'm using a default threshold with a value of 10 for sure. If you agree with me, let me do it.
what do you think?